src/Lexer.php

```diff
@@ -315,8 +315,13 @@ private function lexExpression(): void
             }
         }
 
+        // spread operator
+        if ('.' === $this->code[$this->cursor] && '.' === $this->code[$this->cursor + 1] && '.' === $this->code[$this->cursor + 2]) {
```

Comment:
What happens if you have a `.` near the end of the template ? Would this attempt to access `$this->cursor + 2` past the end of the code ?

Reply:
note that the parsing of arrow functions seem to have the same issue

Reply:
You're totally right - I thought the syntax error due to the missing closing tag took precedence, but it doesn't. It should be covered now.

Reply:
Well, the closing tag is expected *after* lexing the expression, not before it (parsing is done by moving through the file)